pages/form-builder/responses/[[...params]].tsx

```diff
@@ -47,12 +72,87 @@ const Responses: NextPageWithLayout<ResponsesProps> = ({ vaultSubmissions }: Res
     <Button onClick={dialogReportProblemsHandleClose}>{t("responses.reportProblems")}</Button>
   );
 
+  // Note: A hack to enable an ajax file download through axios. Oddly setting the responseType
+  // is not enough. Come back to this, suspect there is a better way..
+  const forceFileDownload = (response: AxiosResponse, fileName: string) => {
+    // NOTE: intentionally allowing Errors to be thrown and caught in container Promise
+    const url = window.URL.createObjectURL(new Blob([response.data]));
+    const link = document.createElement("a");
+    link.href = url;
+    link.setAttribute("download", fileName);
+    document.body.appendChild(link);
+    link.click();
+  };
+
+  const downloadFile = async (url: string, submissionName: string) => {
+    return await axios({
+      url: url,
+      method: "GET",
+      responseType: "blob",
+      timeout: process.env.NODE_ENV === "production" ? 60000 : 0,
+    }).then((response) => {
+      forceFileDownload(response, `${submissionName}.html`);
+    });
+  };
+
+  // NOTE: browsers have different limits for simultaneous downloads. May need to look into
+  // batching file downloads (e.g. 4 at a time) if edge cases/* come up.
+  const handleDownload = async () => {
+    if (getCheckedItems().size === 0) {
+      toast.warn(t("downloadResponsesTable.download.atLeastOneFile"), { position: toastPosition });
+      return;
+    }
+
+    if (getCheckedItems().size > MAX_FILE_DOWNLOADS) {
+      toast.warn(
+        t("downloadResponsesTable.download.trySelectingLessFiles", { max: MAX_FILE_DOWNLOADS }),
+        { position: toastPosition }
+      );
+      return;
+    }
+
+    const toastDownloadingId = toast.info(
+      t("downloadResponsesTable.download.downloadingXFiles", {
+        fileCount: getCheckedItems().size,
+      }),
+      { position: toastPosition, autoClose: false }
+    );
+
+    const downloads = Array.from(getCheckedItems(), (item) => {
+      const url = `/api/id/${formId}/${item[0]}/download`;
```

Comment:
Should check for the value of `item[0]` here vs just trying to use the value --- could cause issues if it's not set for whatever reason